### ISSUE TITLE
Implement part 1 of 2 of the empty bodies in specs/ml/transformer/multi_head_attn.t27 (8 functions)

### ISSUE THREAD

**Issue by gHashTag** [gHashTag/t27#3930]
# Implement part 1 of 2 of the empty bodies in specs/ml/transformer/multi_head_attn.t27 (8 functions)

## Context

`specs/ml/transformer/multi_head_attn.t27` holds 14 functions declared with a body that contains no statement. It is split into 2 issues because one turn cannot carry them all.

**All 2 parts name the same file, so they share one boundary and cannot run at the same time** - the swarm holds the file for whoever has it. They must be done in order, and the check below stops you if you arrived out of turn.

**`.t27` is the hand-authored source language.** `t27c` compiles it out to C, Rust, Verilog and Zig. You are writing source, not compiler output, and not prose about an implementation.

## Before you start

```
$ cd /Users/playom/t27
$ t27c gen specs/ml/transformer/multi_head_attn.t27 2>&1 | grep -c 'not yet implemented'
14      # if it is not 14, this part is not yours yet - stop
```

## What to write

Keep every signature exactly as it is. Quoted verbatim:

1. line 92 - `fn init(config: MHAConfig) -> MHAState`
2. line 101 - `fn forward(state: MHAState, input: []gf16::GF16, mask: AttentionMask) -> MHAForwardResult`
3. line 117 - `fn flash_attention_forward(state: MHAState, input: []gf16::GF16, mask: AttentionMask) -> MHAForwardResult`
4. line 130 - `fn compute_attention_scores(q: []gf16::GF16, k: []gf16::GF16, scale: gf16::GF16) -> []gf16::GF16`
5. line 137 - `fn apply_causal_mask(scores: []gf16::GF16, seq_len: u32) -> []gf16::GF16`
6. line 144 - `fn apply_attention_mask(scores: []gf16::GF16, mask: AttentionMask) -> []gf16::GF16`
7. line 150 - `fn softmax_attention(scores: []gf16::GF16, dim: u32) → []gf16::GF16`
8. line 157 - `fn backward(state: MHAState, result: MHAForwardResult, grad_output: []gf16::GF16) → MHAGradients`

Add a `test` block for each function you implement.

## Acceptance criteria

- 1. `t27c gen specs/ml/transformer/multi_head_attn.t27 > /tmp/t27-gen.zig && grep -c 'not yet implemented' /tmp/t27-gen.zig` prints `6` (on arrival: 14)
- 2. `t27c spec-status specs/ml/transformer/multi_head_attn.t27` does not print `NOPARSE` - the file still parses. (This replaces a parse-metrics criterion. The compiler in the container prints no such metrics, so the command printed nothing and no work could ever satisfy it.)
- 3. with criterion 1 satisfied, all 8 names above still exist: `grep -cE '^\s*(pub )?fn (init|forward|flash_attention_forward|compute_attention_scores|apply_causal_mask|apply_attention_mask|softmax_attention|backward)\(' specs/ml/transformer/multi_head_attn.t27` prints `8`

## User Scenarios & Testing

- **Given** `specs/ml/transformer/multi_head_attn.t27` as it stands today,
  **when** the work described above is done and the commands under Acceptance criteria are run from the repository root,
  **then** each of those commands prints the value stated beside it.

## Requirements

- FR-001: every signature already in the file MUST stay exactly as it is - the signature is the contract, and a caller elsewhere is holding it.
- FR-002: the file MUST still parse afterwards. `t27c parse specs/ml/transformer/multi_head_attn.t27` is the check, and a spec that stops parsing generates nothing, so every test it already carried stops running too.
- FR-003: no function MUST be deleted to satisfy a count.

## Boundary

specs/ml/transformer/multi_head_attn.t27

**Comment by gHashTag** [comment]
Re-filed as https://github.com/gHashTag/t27/issues/4662.

An attempt claimed this issue on 2026-09-21 and left an empty branch. A dispatch row that has spent its retry ceiling is released once and then keeps its issue for good, which is the right rule for a third identical failure and the wrong outcome for an issue nobody ever attempted twice. The new issue carries the same work, the same boundary, and today's brief.

Closing this one so the swarm can take the work again.
